### DISCUSSION TITLE
Improve Documentation : Issues encountered during build on WSL

### DISCUSSION THREAD

**Discussion by 11happy** [organicmaps/organicmaps#6911] [Ideas]
LOCAL ENVIRONMENT-
OS Name	Microsoft Windows 11 Home Single Language
Version	10.0.22621 Build 22621
// I was Trying to build using WSL Ubuntu Focal
Issue Encountered
```
fatal error: ft2build.h: No such file or directory
   34 | #include <ft2build.h>
      |          ^~~~~~~~~~~~
compilation terminated. 
```

Steps I took to resolve this:
I first ran the following command:
```pkg-config --cflags freetype2```
 got the Following output :
 ```
-I/usr/include/freetype2 -I/usr/include/libpng16
```
Now I opened the cmakefile and added the following lines to resolve the issue
```
# Add the FreeType include directory
include_directories(/usr/include/freetype2)
```
Hereby I suggest to also add the solution for above issues in the build section of Installation guide. So I ask whether to open an issue? 
Thank you.

**Comment by biodranik** [comment]
No workarounds, please. Freetype is already included by CMake, you need to debug why it doesn't work.

```
# Should be on the root level, not in 3party, so tests can get these dependencies.
# Should go before 3party as harfbuzz is using them.
if (LINUX_DETECTED)
  find_package(ICU COMPONENTS uc i18n data REQUIRED)
  find_package(Freetype REQUIRED)
endif()

```

**Reply by 11happy** [reply]
To solve it in a more general sense I think after  this line
```
find_package(Freetype REQUIRED)
```
we can use this code to ensure this error doesn't occurs
```
include_directories(${FREETYPE_INCLUDE_DIRS})
```

**Reply by 11happy** [reply]
Currently I am also trying to find the root cause of the issue . Also if there is any guidance for me I am open to it .
Thank you

**Reply by biodranik** [reply]
What about this? https://cmake.org/cmake/help/latest/module/FindFreetype.html

If explicit includes are requred, then it's better to use target_include_directories

**Reply by 11happy** [reply]
Yes, How about this?
```
if(FREETYPE_FOUND)
 target_include_directories(omim PRIVATE ${FREETYPE_INCLUDE_DIRS})
else()
 message(FATAL_ERROR "FreeType not found. Please install it.")
endif()
```
Thank you

**Reply by biodranik** [reply]
Why omim target? Freetype is used by Drape only, so this if should be used there.

**Reply by 11happy** [reply]
Certainly I agree with you, I am sorry for my mistake and am trying to understand the codebase better. Also How should I proceed next?
Thank you

**Reply by biodranik** [reply]
Make a fix, test it with and without `-DWITH_SYSTEM_PROVIDED_3PARTY=on` defined, and create a PR.

**Reply by 11happy** [reply]
okay will be doing this.

**Reply by 11happy** [reply]
Just an update on the task : Currently I have some important commitments for next 2 days so I will start working on it from 24 and will make a PR as early as possible.
Thank you

**Comment by 11happy** [comment]
I attempted to investigate the issue thoroughly and sought to identify its root cause. To do this, I performed a fresh installation of WSL with Ubuntu 20.04.6 LTS focal and attempted to reproduce the error. However, I did not encounter any issues during the installation process. Subsequently, I repeated the process with a fresh installation of WSL using Ubuntu 22.04.3 LTS Jammy, and once again, I could not reproduce the error that I had previously encountered.

As a result, I believe that the earlier issue may have been caused by some compiler-specific(not able to link that particular library) problem in my previous WSL Ubuntu setup.
I am open for any guidance related to this matter, However I think it is not reproducible and will try to take some good first issues.
Thank you
